SwiftCompilerSources/Sources/Optimizer/ModulePasses/MandatoryPerformanceOptimizations.swift

```diff
@@ -123,9 +123,44 @@ private func shouldInline(apply: FullApplySite, callee: Function) -> Bool {
     // Force inlining them in global initializers so that it's possible to statically initialize the global.
     return true
   }
+  if apply.parentFunction.isGlobalInitOnceFunction, let global = getGlobal(of: apply.parentFunction), global.mustBeInitializedStatically {
+    if resultOfCalleeAssignsAMember(apply: apply) {
+      return true
+    }
+  }
   return false
 }
 
+private func resultOfCalleeAssignsAMember(apply: FullApplySite) -> Bool {
```

Comment:
If we want to support things like
```
init() {
  let x = X()
  self.a = x
  self.b = x
}
```
Then a loop is not sufficient. It needs to be a recursive function, visiting _all_ uses rather than the singleUse.